pkg/chunk/backoff.go

```diff
@@ -0,0 +1,44 @@
+package chunk
```

Comment:
I really like this abstraction. No action required.

Reply:
Theres a similar one here, could we perhaps unify them? https://github.com/weaveworks/cortex/blob/3c941b945c580741f1df6c36f917164b11be9f86/pkg/ring/consul_client.go#L196

Reply:
Very good point; I raised a separate issue because I'd like to merge this before doing that.